src/librustc/middle/stability.rs

```diff
@@ -509,11 +525,16 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         }
     }
 
-    pub fn check_stability(self, def_id: DefId, id: NodeId, span: Span) {
+    /// Evaluates the stability of an item.
+    ///
+    /// Returns `None` if the item is stable, or unstable but the corresponding `#![feature]` has
+    /// been provided. Returns the tuple `Some((feature, reason, issue))` of the offending unstable
+    /// feature otherwise.
+    pub fn eval_stability(self, def_id: DefId, id: NodeId, span: Span) -> EvalResult {
```

Comment:
Pre-existing, but can we add a comment to the `id` parameter -- from what I can tell, if it is `DUMMY_NODE_ID`, that means we suppress deprecation lints (and otherwise we attach them to that `id`).

Personally, I think we should go further and change this to `Option<NodeId>` instead of using `DUMMY_NODE_ID` -- and if you want to do that it will make me happy -- but I won't require it. =)

Reply:
Changed to `Option<NodeId>` in the coming commits :)